### PR TITLE
better document cron options

### PR DESCRIPTION
**Pull request by sw-yx**
## What does this PR do?

image

better document cron options. 

- https://deploy-preview-439--mystifying-fermi-1bc096.netlify.app/docs/go/distributed-cron
- https://deploy-preview-439--mystifying-fermi-1bc096.netlify.app/docs/java/distributed-cron
- https://deploy-preview-439--mystifying-fermi-1bc096.netlify.app/docs/php/distributed-cron

## Notes to reviewers

i am moving some of the cron content into shared content across the 3 languages, and making them more discoverable from each SDK's workflows page.

i dont think node sdk has cron yet so am leaving that for now

### CODE REVIEW

File: docs/php/workflows.md [hunk 1]
```diff
@@ -323,6 +324,10 @@ $run = $workflowClient->start($accountTransfer, 'fromID', 'toID', 'refID', 1000)
 var_dump($run->getExecution()->getID());
 ```
 
+### Recurring start
+
+You can start your workflows on a regular schedule with [the CronSchedule option](distributed-cron).
```
Comment (flossypurse) [inline docs/php/workflows.md:329]:
```suggestion
You can start a Workflow Execution on a regular schedule with [the CronSchedule option](distributed-cron).
```

File: docs/php/workflows.md [hunk 2]
```diff
@@ -42,12 +42,12 @@ Workflows can also answer synchronous [Queries](/docs/php/queries) and receive [
 All interface methods must have one of the following annotations:
 
 - **#[WorkflowMethod]** indicates an entry point to a Workflow.
-It contains parameters such as timeouts and a task queue.
-Required parameters (such as `executionStartToCloseTimeoutSeconds`) that are not specified through the annotation must be provided at runtime.
+  It contains parameters such as timeouts and a task queue.
```
Comment (flossypurse) [inline docs/php/workflows.md:45]:
```suggestion
  It contains parameters that specify timeouts and a Task Queue name.
```

File: docs/java/workflows.md [hunk 3]
```diff
@@ -356,6 +356,10 @@ A Synchronous start initiates a Workflow and then waits for its completion. The
 <!--SNIPSTART hello-world-project-template-java-workflow-initiator-->
 <!--SNIPEND-->
 
+### Recurring start
+
+You can start your workflows on a regular schedule with [the CronSchedule option](distributed-cron).
```
Comment (flossypurse) [inline docs/java/workflows.md:361]:
```suggestion
You can start a Workflow Execution on a regular schedule with [the CronSchedule option](distributed-cron).
```

File: docs/go/workflows.md [hunk 4]
```diff
@@ -155,7 +155,8 @@ we.Get(ctx, &result)
 
 :::note
 
-In most uses cases it is better to be prepared to execute the Workflow asynchronously.
+In most use cases it is better to execute the Workflow asynchronously.
+You can also start your workflows on a regular schedule with [the CronSchedule option](distributed-cron).
```
Comment (flossypurse) [inline docs/go/workflows.md:159]:
```suggestion
You can also start a Workflow Execution on a regular schedule with [the CronSchedule option](distributed-cron).
```

File: docs/shared/distributed-cron.md [hunk 5]
```diff
@@ -0,0 +1,42 @@
+You can turn any Temporal Workflow into a repeatedly initiated Workflow. Two choices:
```
Comment (flossypurse) [inline docs/shared/distributed-cron.md:1]:
```suggestion
There are two ways that you can turn any Temporal Workflow into a repeatedly executed Workflow.
```

File: docs/shared/distributed-cron.md [hunk 6]
```diff
@@ -0,0 +1,42 @@
...
+
+- start a Workflow using the Temporal CLI with an optional cron schedule using the `--cron` argument
```
Comment (flossypurse) [inline docs/shared/distributed-cron.md:3]:
```suggestion
1. Start a Workflow Execution using the Temporal CLI with an optional cron schedule using the `--cron` argument.
```

File: docs/shared/distributed-cron.md [hunk 7]
```diff
@@ -0,0 +1,42 @@
...
+- (recommended) Supply a cron schedule when starting the Workflow using the CronSchedule
```
Comment (flossypurse) [inline docs/shared/distributed-cron.md:4]:
```suggestion
2. **Recommended**: Supply a cron schedule when starting the Workflow Execution using the CronSchedule.
```

I know you like to front load recommendations, which I support. I just want to find a way for us to do it that looks a bit cleaner such that we can build style guidance around it.
I don't want to block on this one, as I need to think about how to achieve this long term.

File: docs/shared/distributed-cron.md [hunk 8]
```diff
@@ -0,0 +1,42 @@
...
+  parameter of <a href={props.docUrl}>StartWorkflowOptions</a>.
+
+<div>{props.children}</div>
+
+For Workflows with CronSchedule:
+
+- CronSchedule is based on UTC time. For example cron schedule "15 8 \* \* \*"
```
Comment (flossypurse) [inline docs/shared/distributed-cron.md:11]:
```suggestion
- CronSchedule is based on UTC time.
For example, cron schedule "15 8 \* \* \*" will run daily at 8:15am UTC.
```

File: docs/shared/distributed-cron.md [hunk 9]
```diff
@@ -0,0 +1,42 @@
...
+  will run daily at 8:15am UTC.
```
Comment (flossypurse) [inline docs/shared/distributed-cron.md:12]:
```suggestion
```

File: docs/shared/distributed-cron.md [hunk 10]
```diff
@@ -0,0 +1,42 @@
...
+- If a Workflow failed and a RetryPolicy is supplied to the StartWorkflowOptions
```
Comment (flossypurse) [inline docs/shared/distributed-cron.md:13]:
```suggestion
- If a Workflow Execution failed and a RetryPolicy is supplied to the `StartWorkflowOptions`, the Workflow Execution will be retried based on the RetryPolicy.
```

File: docs/shared/distributed-cron.md [hunk 11]
```diff
@@ -0,0 +1,42 @@
...
+  as well, the Workflow will retry based on the RetryPolicy. While the Workflow is
```
Comment (flossypurse) [inline docs/shared/distributed-cron.md:14]:
```suggestion
While the Workflow Execution is retrying, the Server will not schedule the next Workflow Execution.
```

File: docs/shared/distributed-cron.md [hunk 12]
```diff
@@ -0,0 +1,42 @@
...
+  retrying, the server will not schedule the next cron run.
```
Comment (flossypurse) [inline docs/shared/distributed-cron.md:15]:
```suggestion
```

File: docs/shared/distributed-cron.md [hunk 13]
```diff
@@ -0,0 +1,42 @@
...
+- Temporal server only schedules the next cron run after the current run is
```
Comment (flossypurse) [inline docs/shared/distributed-cron.md:16]:
```suggestion
- The Temporal Server only schedules the next Workflow Execution after the current execution has completed.
```

File: docs/shared/distributed-cron.md [hunk 14]
```diff
@@ -0,0 +1,42 @@
...
+  completed. If the next schedule is due while a Workflow is running (or retrying),
```
Comment (flossypurse) [inline docs/shared/distributed-cron.md:17]:
```suggestion
If the next execution is due to occur while the Workflow is currently executing (including retries), then the next execution will be skipped.
```

File: docs/shared/distributed-cron.md [hunk 15]
```diff
@@ -0,0 +1,42 @@
...
+  then it will skip that schedule.
```
Comment (flossypurse) [inline docs/shared/distributed-cron.md:18]:
```suggestion
```

File: docs/shared/distributed-cron.md [hunk 16]
```diff
@@ -0,0 +1,42 @@
...
+- Cron Workflows will not stop until they are terminated or cancelled.
```
Comment (flossypurse) [inline docs/shared/distributed-cron.md:19]:
```suggestion
- Cron initiated Workflow Executions will not stop until they are terminated or cancelled.
```

File: docs/java/workflows.md [hunk 17]
```diff
@@ -327,7 +327,7 @@ This is the most common way to start Workflows in a live environment.
 
 If you need to wait for the completion of a Workflow after an asynchronous start, the most straightforward way is to call the blocking Workflow instance again.
 
-If `WorkflowOptions.WorkflowIdReusePolicy` is not set to `AllowDuplicate`, then instead of throwing `DuplicateWorkflowException`, 
+If `WorkflowOptions.WorkflowIdReusePolicy` is not set to `AllowDuplicate`, then instead of throwing `DuplicateWorkflowException`,
 it reconnects to an existing Workflow and waits for its completion.
 
 The following example shows how to do this from a different process than the one that started the Workflow.
```
Comment (flossypurse) [inline docs/java/workflows.md:333]:
```suggestion
The following example shows how to do this from a different process than the one that started the Workflow Execution.
```

**Review by flossypurse — CHANGES_REQUESTED** [review]

**Review by flossypurse — APPROVED** [review]

**Comment by sw-yx** [comment]
Here is a test comment

screenshot


<details><summary>Browser metadata</summary>





```
Path:      /blog
Browser:   Chrome 90.0.4430.212 on Mac OS 10.15.7
Viewport:  1920 x 1000 @1x
Language:  en-US
Cookies:   Enabled
```

[View on BrowserStack](https://live.browserstack.com/dashboard#url=https%3A%2F%2Fdeploy-preview-439--mystifying-fermi-1bc096.netlify.app%2Fblog&os=OS+X&os_version=Catalina&browser=Chrome&browser_version=90.0&resolution=1920x1000&speed=1&start=true)



</details>

[Open Deploy Preview](https://deploy-preview-439--mystifying-fermi-1bc096.netlify.app/blog) · [Mark as Resolved](https://app.netlify.com/cdp/resolve?deployID=60a52d8bad32bd00072a55ee&commentID=60a53ed1e33f0d175ee567e3&resolution=resolved)

**Pull request merged**